app/models/user.rb

```diff
@@ -32,11 +35,36 @@ def self.search(input)
     all
   end
 
+  # Verifies the token existence, fetches its user and validates its expiration
+  # and invalidates the user token if expired.
+
+  def self.verify_reset_token!(token)
```

Comment:
You can drop the `!`

Reply:
I'm using that as a convention just like with models'query methods.
It indicates that the method uses at least one query method that raises an exception.

I think since we want the validation operation to be silent using query methods that raises exceptions is prohibited.
Correct?